### PR TITLE
Commandline and other documentation updates + configuration class + duration fixes

### PR DESCRIPTION
**Pull request by goodwillcoding**
* Major updates to commandline documentation, documenting all the flags
* fix inclusion of module documentation
* document run state result
* fix documentation for MatheExpression (* special char was needed escaping)
* document usage of world and world.config
* add Configuration class and it's tests to replace world.config being a
  lambda function, which was very odd to print out when dropped into debugging
  of steps
* for bdd xml and cucumber json fix duration to not show in scientific notation
  but instead it is not now a float rounded up to 10 decimal point.
* rename dry_run in runner.py to show_only since it was conflicting with the
  config.dry_run that was set by --dry-run command line switch
* fix small typos in api docs
* gitignore the vendored modules

### CODE REVIEW

File: docs/commandline.rst [hunk 1]
```diff
@@ -1,92 +1,358 @@
...
-This chapter describes how to use radish from the command line. All it's commands, options and arguments.
+This chapter describes how to use Radish from the command line. All it's
+commands, options and arguments.
+
 
 Run - Specify Feature files
 ---------------------------
 
-All arguments which doesn't belong to any command line switch are interpreted as Feature files or Feature file locations. If the argument is a directory all files ending with ``.feature`` will be run. It's possible to mix files and directories:
+All arguments which doesn't belong to any command line switch are interpreted
```
Comment (timofurrer) [inline docs/commandline.rst]: `don't` instead of `doesn't`

File: docs/commandline.rst [hunk 2]
```diff
@@ -1,92 +1,358 @@
...
   radish SomeFeature.feature myfeatures/
 
+
 Run - Specify base directory
 ----------------------------
 
-The directory where the *Step* and *Terrain* files are located is called the ``base directory``.
-Per default it points to ``$PWD/radish``. All python files within this directory are imported by radish. You can specify the base directory with the ``-b`` or ``--basedir`` switch:
+Radish looks for *Step* and *Terrain* files in the ``base directory``. By
+default the ``base directory`` is set to the in the ``radish`` folder within
```
Comment (timofurrer) [inline docs/commandline.rst]: Delete redundant `the in`

File: docs/commandline.rst [hunk 3]
```diff
@@ -1,92 +1,358 @@
...
+the current working directory (a.k.a ``$PWD/radish``). Radish imports all
+the python files imported.
```
Comment (timofurrer) [inline docs/commandline.rst]: Remove redundant `imported` in the end of the sentence and reword sentence a little

File: docs/commandline.rst [hunk 4]
```diff
@@ -1,92 +1,358 @@
...
+-----------------
+
+Radish provides ability to debug each step using an IPython debugger. You can
+enable that using ``--debug-steps`` command line switch.
+
+.. code:: bash
+
+  radish --debug-steps SomeFeature.feature
+
+If you are unfamiliar with the Python debugger please consult official
```
Comment (timofurrer) [inline docs/commandline.rst]: Maybe we should mention that the python `pdb` is just a fallback if the IPython pdb cannot be found.

File: docs/commandline.rst [hunk 5]
```diff
@@ -1,92 +1,358 @@
...
+
+  radish SomeFeature.feature -t
+  radish SomeFeature.feature --with-traceback
+
+
+Run - Use custom marker to uniquely identify test run
+-----------------------------------------------------
+
+Radish supports marker functionality which is used to uniquely identify a
+specific test run using. By default the marker is set to the number of seconds
```
Comment (timofurrer) [inline docs/commandline.rst]: The `using` doesn't not make sense here.

File: docs/commandline.rst [hunk 6]
```diff
@@ -1,92 +1,358 @@
...
+.. code:: bash
+
+  radish SomeFeature.feature -m "My Marker"
+  radish SomeFeature.feature --marker "My Marker"
+
+  ... radish output
+
+  Run My Marker finished within 0:0.001272 minutes
+
+The marker is also passed in to all the hooks define in terrain file. To see
```
Comment (timofurrer) [inline docs/commandline.rst]: `defined in the terrain file` would sound better.

File: docs/commandline.rst [hunk 7]
```diff
@@ -1,92 +1,358 @@
...
+
+Radish allows you to pass custom data to Terrain hook code or to Steps code
+using the ``-p`` or ``--profile`` is a command line switch. This can be used to
+customize your test runs as needed.
+
+The value specified to the ``-p`` / ``--profile`` command line switch is made
+available in ``world.config.profile``. Please see :ref:`tutorial#world` for
+for an example.
+
+A common usage of  is to set it to the environment value
```
Comment (timofurrer) [inline docs/commandline.rst]: `usage of this is`

File: docs/commandline.rst [hunk 8]
```diff
@@ -1,92 +1,358 @@
...
   radish SomeFeature.feature -d
   radish SomeFeature.feature --dry-run
 
-Run - Specify certain Scenarios by id
--------------------------------------
+Run - Specifying Scenarios by id
+--------------------------------
+
+Radish can also runs specific scenarios by id using the ``-s`` or
+``--scenarios`` command line switch. The ids are scenarios are indexed by the
```
Comment (timofurrer) [inline docs/commandline.rst]: The second `are` is redundant.

File: docs/commandline.rst [hunk 9]
```diff
@@ -1,92 +1,358 @@
...
 
-Run - Specify certain Features/Scenarios by tags
-------------------------------------------------
 
-radish is able to run only a selection of certain Features and/or Scenarios. The Features/Scenarios must be tagged. Use the ``--feature-tags`` or ``--scenario-tags`` to specify the tags of Features/Scenarios which should be run. The value can be a single tag or a comma separated list of tags:
+Run - Shuffle Scenarios
+-----------------------
+
+Radish can also shuffle the Scenarios by using the ``--shuffle`` command line
+switch. This useful when you are trying to detect if any scenario have
```
Comment (timofurrer) [inline docs/commandline.rst]: `This is useful`

File: docs/commandline.rst [hunk 10]
```diff
@@ -1,92 +1,358 @@
...
+
+Radish can write out a Cucumber style JSON file with the results of after a
+test run using ``--cucumber-json`` command line switch. The command line switch
+value must be a file path to the output file.
+
+.. code:: bash
+
+  radish SomeFeature.feature --cucumber-json /tmp/result.json
+
+Documentation describing then format of the Cucumber JSON file can be founde
```
Comment (timofurrer) [inline docs/commandline.rst]: `the format`

File: docs/commandline.rst [hunk 11]
```diff
@@ -1,92 +1,358 @@
...
+turned of using ``--write-steps-once``.
+
+.. code:: bash
+
+  radish SomeFeature.feature --no-ansi
+  radish SomeFeature.feature --no-ansi --write-steps-once
+
+The ``--no-line-jump`` command line switch disables the "overwriting" of the
+yellow pending lines by the success or failure lines. This helpful to you
+during debugging so you can see when steps were pending then executed.
```
Comment (timofurrer) [inline docs/commandline.rst]: Maybe mention `print debugging` here?

**Comment by goodwillcoding** [comment]
@timofurrer: all the feedback has been addressed. Sorry for the errors, hard to see my own typos after a lot of writing.

**Comment by timofurrer** [comment]
Thanks! :beers: :cake:

**Pull request merged**